transport-native-epoll/src/main/c/io_netty_channel_epoll_Native.c

```diff
@@ -1348,3 +1349,101 @@ JNIEXPORT jstring JNICALL Java_io_netty_channel_epoll_Native_strError(JNIEnv* en
     char* err = strerror(error);
     return (*env)->NewStringUTF(env, err);
 }
+
+JNIEXPORT jint JNICALL Java_io_netty_channel_epoll_Native_socketDomain(JNIEnv* env, jclass clazz) {
+    int fd = socket(PF_UNIX, SOCK_STREAM | SOCK_NONBLOCK, 0);
+    if (fd == -1) {
+        return -errno;
+    }
+    return fd;
+}
+
+JNIEXPORT jint JNICALL Java_io_netty_channel_epoll_Native_bindDomainSocket(JNIEnv * env, jclass clazz, jint fd, jstring socketPath) {
+    struct sockaddr_un addr;
+
+    memset(&addr, 0, sizeof(addr));
+    addr.sun_family = AF_UNIX;
+
+    const char *socket_path = (*env)->GetStringUTFChars(env, socketPath, 0);
+    memcpy(addr.sun_path, socket_path, strlen(socket_path));
+
+    if (unlink(socket_path) == -1 && errno != ENOENT) {
+        return -errno;
+    }
+
+    int res = bind(fd, (struct sockaddr*) &addr, sizeof(addr));
+    (*env)->ReleaseStringUTFChars(env, socketPath, socket_path);
+
+    if (res == -1) {
+        return -errno;
+    }
+    return res;
+}
+
+JNIEXPORT jint JNICALL Java_io_netty_channel_epoll_Native_connectDomainSocket(JNIEnv * env, jclass clazz, jint fd, jstring socketPath) {
+    struct sockaddr_un addr;
+
+    memset(&addr, 0, sizeof(addr));
+    addr.sun_family = AF_UNIX;
+
+    const char *socket_path = (*env)->GetStringUTFChars(env, socketPath, 0);
```

Comment:
I wonder if we should convert this to the current system character encoding, because not all systems use UTF-8 as the system encoding.